### ISSUE TITLE
button.replied is returned false even if replied (in message component collector)

### ISSUE THREAD

**Issue by amrit-2006** [discordjs/discord.js#8048]
### Which package is this bug report for?

discord.js

### How to reproduce the bug
**Procedure**

The button is replied(ephemeral) in the interactionCreate Event. The button is then deferred in the Collector Event. 

**Explaination** 

Here we know that the button is `replied` in the collector event. Hence I added a `if statement` checking if the button is replied. The button is supposed to be deferred at the second time only if the button is not replied. This means that, According to the data from the collector event, the button is not replied. Btw the collector is a `message.createMessageComponentCollector()`. 

### Extra Information

The buttons logs `button.replied`  as `true`. But it logs as `false` inside the Collector Event. I am not very sure if `button.replied` is supposed to be updated in Collector event. But I think it is possible since most of the DJS functions return a Promise.

I am also not sure if this is a bug or if it's just that the code does not work for me alone

### Code sample

```typescript
await button.reply({content:'test',ephemeral:true}) => Interaction Event 
if (!button.replied && !button.deferred) {await button.deferUpdate()} -> Collector event
```

### Package version

13.8

### Node.js version

16.9

### Operating system

Windows

### Priority this issue should have

Medium (should be fixed soon)

### Which partials do you have configured?

Channel, GuildMember, Message, Reaction

### Which gateway intents are you subscribing to?

Guilds, GuildMembers, GuildMessages, DirectMessages

**Comment by amrit-2006** [comment]
console.log of `button.replied` is shown as true only after the `button.reply` in interaction event

**Comment by monbrey** [comment]
Any button interactions received by the collector would be entirely new interactions that you have not replied to yet.

If you have both an interactionCreate event and a collector listening to the same button, then you likely have a race condition and are risking duplicate responses.

The code you have provided is not a reproducible code sample - reproducible means being able to copy and paste it into a blank bot, run it, and get the same result. Could you please provide the necessary context for those two lines?

**Comment by MeXi-001** [comment]
@amrit-2006 You can use `<ButtonInteraction>.fetchReply()` in the collector to find out whether the interaction was already replied to or not
